cli/src/Plugin.ts

```diff
@@ -0,0 +1,61 @@
+import { ChildProcess } from "./ChildProcess";
+import { Diagnostic } from "./Diagnostic";
+import { Entity } from "./Entity";
+import { Page } from "./Page";
+
+/**
+  Converts docs generator tool output to Yokedox-friendly output.
+ */
+export type Plugin = {
+  /**
+    Entrypoint for plugin.
+    
+    Throw Error to report errors.
+   */
+  run(args: PluginArgs): Promise<void> | void;
+};
+
+/**
+  Passed to a plugin's run() function.
+ */
+export type PluginArgs = {
+  /**
+    The generator executable path.
+   */
+  generator: string;
+
+  /**
+    The path to the temporary working directory. Plugins may freely write within
+    this directory, but should not delete the directory itself. No cleanup
+    necessary.
+   */
+  tempDir: string;
+
+  /**
+    The interface to the shell. Plugins should use this instead of their own
+    imports of child_process.
+   */
+  child_process: ChildProcess;
```

Comment:
Why the indirection here?

Reply:
I was thinking this could be mockable for unit testing (similar to memfs for Bluehawk), but it may not be worth it.